packages/dds/tree/src/edit-manager/editManager.ts

```diff
@@ -128,44 +132,65 @@ export class EditManager<TChangeset, TChangeFamily extends ChangeFamily<any, TCh
             return commitToRebase.changeset;
         }
 
+        let nextRevisionNum = -1;
         const changeRebasedToRef = branch.localChanges.reduceRight(
             (newChange, branchCommit) =>
                 this.changeFamily.rebaser.rebase(
                     newChange,
-                    this.changeFamily.rebaser.invert(branchCommit.changeset),
+                    {
+                        revision: brand(nextRevisionNum--),
+                        change: this.changeFamily.rebaser.invert(changeWithMetadataFromCommit(branchCommit)),
+                    },
                 ),
             commitToRebase.changeset,
         );
 
-        return this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+        const fullyRebasedChange = this.rebaseOverCommits(changeRebasedToRef, this.getCommitsAfter(branch.refSeq));
+
+        const shouldRemoveReference = (revision: RevisionTag): boolean =>
+            isTemporaryRevision(revision) || isBranchLocalRevision(branch, revision);
+
+        return this.changeFamily.rebaser.filterReferences(
+            fullyRebasedChange,
+            shouldRemoveReference,
+        )
     }
 
     // TODO: Try to share more logic between this method and `rebaseBranch`
-    private rebaseLocalBranch(trunkChange: TChangeset): TChangeset {
-        const newBranchChanges: TChangeset[] = [];
-        const inverses: TChangeset[] = [];
+    private rebaseLocalBranch(trunkChange: ChangeWithMetadata<TChangeset>): TChangeset {
+        const newBranchChanges: ChangeWithMetadata<TChangeset>[] = [];
+        const inverses: ChangeWithMetadata<TChangeset>[] = [];
 
+        let nextTempRevision = -1;
         for (const localChange of this.localChanges) {
             let change = this.rebaseChange(localChange, inverses);
             change = this.changeFamily.rebaser.rebase(change, trunkChange);
             change = this.rebaseChange(change, newBranchChanges);
 
-            newBranchChanges.push(change);
+            newBranchChanges.push({ revision: brand(nextTempRevision--), change });
+
+            const inverse = this.changeFamily.rebaser.invert({
+                revision: brand(nextTempRevision--),
+                change: localChange,
+            });
 
-            inverses.unshift(this.changeFamily.rebaser.invert(localChange));
+            inverses.unshift({ revision: brand(nextTempRevision--), change: inverse });
```

Comment:
Shouldn't this be
```suggestion
            inverses.unshift({ revision: brand(nextTempRevision), change: inverse });
```

Reply:
If so, did test coverage not catch that because those numbers don't matter so long as all the local changes are rebased over the same set?

Reply:
Oh wait no, I missread the code for `inverse`. Closing this.